daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -376,16 +376,13 @@ private[lf] final class Compiler(
         )
       case structProj: EStructProj =>
         structProj.fieldIndex match {
-          case None =>
-            throw CompilationError(
-              s"structural record projection for field ${structProj.field} has no index")
+          case None => SBStructProjByName(structProj.field)(compile(structProj.struct))
           case Some(index) => SBStructProj(index)(compile(structProj.struct))
         }
```

Comment:
Why not something like: 

```scala
   val builtin = structProj.fieldIndex.fold(SBStructProjByName(structProj.field))(SBStructProj)
   builtin.apply(compile(structProj.struct))
```

Reply:
I simply don't like `fold` on `Option`. Whenever I see `fold`, I need to stop for a moment and run a type inference in my head. I don't need to do this with pattern matching. Thus, I'm avoiding to produce code I _personally_ find harder to read, although I won't stop others from using `fold`.

Reply:
No problem, it was just a suggestion.

Reply:
I know. Apologies if I came across as aggressive!

My goal was to share my motivation for doing things the way I do them. 😃